### ISSUE TITLE
An unexpected error occurred - "connect ENOENT \\.\pipe\errorReporter"

### ISSUE THREAD

**Issue by davidilag** [docker/for-win#14191]
### Description

Installed this docker installation file to Windows 11 that has WSL 2:
https://desktop.docker.com/win/main/amd64/Docker%20Desktop%20Installer.exe?utm_source=docker&utm_medium=webreferral&utm_campaign=docs-driven-download-win-amd64&_gl=1*15pg1hk*_gcl_au*MzIyNDA0NTE4LjE3MTQyMzc5Njg.*_ga*MTc4NDM0MTEwNi4xNzE0MjM3OTY4*_ga_XJWPQMJYHQ*MTcyMDc5NDQ1NS41LjEuMTcyMDc5NTQ1MC42MC4wLjA.

### Reproduce

1. install Docker 
2. open Docker Desktop app

### Expected behavior

That Docker Desktop app starts

### docker version

```bash
Docker Desktop 4.29.0
```

### docker info

```bash
connect ENOENT \\.\pipe\errorReporter
```

### Diagnostics ID

7F5B1822-DE06-42EE-8569-A164E58F027E/20240712144537

### Additional Info

https://docker-pinata-support.s3.amazonaws.com/incoming/3/7F5B1822-DE06-42EE-8569-A164E58F027E/20240712144537.zip

**Comment by ahmad-alismail-xax** [comment]
Same issue here
![image](https://github.com/user-attachments/assets/0072b91f-8160-4cc3-abc2-6b01f7515e40)

Have you found any solution?

**Comment by ahmad-alismail-xax** [comment]
Same issue here:

Diagnostics ID:     3E31265E-34D8-48A7-913D-C7C649C235AD/20240715135441

Uploaded diagnostics to: https://docker-pinata-support.s3.amazonaws.com/incoming/3/3E31265E-34D8-48A7-913D-C7C649C235AD/20240715135441.zip

**Comment by jpbriend** [comment]
Can you add your Windows user to the Windows group named `docker-users`?
Following https://docs.docker.com/desktop/release-notes/#for-windows-8

**Comment by davidilag** [comment]
@jpbriend , thanks. This gave me what I needed to proceed with installing Docker-Desktop. 

I have two Windows users at my work, one normal and one admin. I am logged in with the normal Windows user where I am trying to install Docker. The normal user is not administrator on the computer, only the admin user. U added the normal Windows user to the Windows group called _docker-users_ from the terminal as suggested @jpbriend. I also needed to start the Docker-Desktop app as admin.

The next error I got was related to what Docker CLI context was used. A quick ChatGPT search gave me this solution, run it in the terminal (CMD) as administrator:

- _docker context ls_ (lists the available docker contexts. I only had a default one, and the linux was missing)
- _docker context create desktop-linux --docker "host=npipe:////./pipe/docker_engine"_ (this creates a linux context called desktop-linux)
- _docker context use desktop-linux_ (this switches to the newly created linux context)
- _docker info_ (checks that the docker daemon is running)

Now Docker-Desktop is staring and the "welcome-to-dock" docker container is working (a simple web app with a welcome page). But it seems that I cannot add and run new images/containers. 

I am getting this error after the other stuff when running the last command:
_Server:
ERROR: error during connect: this error may indicate that the docker daemon is not running: Get "http://%2F%2F.%2Fpipe%2Fdocker_engine/v1.45/info": open //./pipe/docker_engine: Access is denied.
errors pretty printing info_

ChatGPT suggests that the Docker Desktop Service Windows service is not running, which it was not. I started it, restarted the Docker-desktop app, but I still get this error. There are other suggestions to what the problem could be, some related to the fact that my normal Windows user is not administrator on the laptop I am installing Docker-desktop on. **This seems to be the root problem.**

I have now made my normal Windows user to admin using my admin user (in the terminal). Now things are behaving normally. 

Form the terminal, after logging in to docker with the command _docker login_ and downloading this repo: _git clone https://github.com/docker/welcome-to-docker_ (Git for Windows needs to be installed),  I am now able to run this command:

- docker build -t welcome-to-docker .

This adds the docker container to the list of images in the Docker-desktop app where I am able to start it and run the application.

**Comment by jpbriend** [comment]
I also agree it's an issue with your non-admin user not being admin.
After adding your non-admin user to the `docker-users` group, did you restarted your Windows session?
I suspect the group will be given to the user only on a new session.

**Comment by davidilag** [comment]
Hi @jpbriend 
I believe I did a Windows restart after this step - I restarted multiple times along the way. This is a typical Windows thing when settings like this are updated. However, it did not solve all the problems, but was one of the settings that needed to be made. I have added this step in my explanation above.

**Comment by chelnak** [comment]
👋 For some more context:

You may see the error message above when Docker Desktop cannot read the named pipe used for reporting errors. In most cases this is usually due to a group membership issues (as mentioned above).

We've made the default error dialog a bit more descriptive now, so it should give more of an indication of what needs to be done to resolve the issues.

**Comment by davidilag** [comment]
Hi @chelnak 

After the process of finding out what the problem was and solving it, the initial error message kind of make sense. In the beginning it did not.

Prior to be able to install Docker Desktop, I had an issue with WSL 2 (Linux with Windows) on my new Windows 11 laptop. This is a general problem it seems. After some time googling that error I managed to get the WSL 2 working and a Ubuntu 22.04.3 LTS instance running. After that I got the error message that is described in this issue.
